### PR TITLE
[ISSUE #72] LinkedIn Ads documentation

### PR DESCRIPTION
**Pull request by maxi297**
## What
Addresses https://github.com/airbytehq/snowflake-native-apps/issues/72

## How
Taking the process extracted from https://github.com/airbytehq/snowflake-native-apps/issues/71 and creating a new document describing the process. 

This document should not be shown in the sidebar as it is not defined in https://github.com/airbytehq/airbyte/blob/master/docusaurus/sidebars.js. This should make it hidden but still accessible through https://docs.airbyte.com/snowflake-native-apps/linkedin-ads

### CODE REVIEW

File: docs/snowflake-native-apps/linkedin-ads.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+9. Wait for the application to install. Once complete, the pop-up window should automatically close.
+
+You should now see the application under `Installed Apps`. You may need to refresh the page.
+
+## LinkedIn Account
+In order for the Snowflake Native App to query LinkedIn, you will need an account with the right permissions. Please follow [the LinkedIn Ads authentication guide](https://docs.airbyte.com/integrations/sources/linkedin-ads/#set-up-linkedin-ads-authentication-airbyte-open-source) for further information.
+
+## Snowflake Native App Authorizations
+> **Note**  
+> All the queries below use `LINKEDIN_ADS_TEST` but if you have renamed your application, you will have to use this name as a reference.
```
Comment (maxi297) [inline docs/snowflake-native-apps/linkedin-ads.md:35]: I added this line based on [this line](https://github.com/airbytehq/airbyte/pull/30386/files#diff-9199f80dfa9350b869d60d63787fea386c7146e37b9600d30961d174066235c2R23) but I haven't tested this

Comment (brianjlai) [reply docs/snowflake-native-apps/linkedin-ads.md:35]: i think we should do some sort of bracketing like `<your db>` or something, but LINKEDIN_ADS_TEST is not too nice to see, i'll update relevant parts

**Comment by brianjlai** [comment]
@maxi297 a few notes on changes I pushed to the branch

- I modified public -> your_schema. Though public is common, I think using your_schema is more accurate because a customer can define their own schemas. And also we did reference schema a few times so better for consistency
- Added the additional step we need which is to grant CREATE to the app so that it can create the tables if they don’t already exist
- Renamed the app AIRBYTE_LINKEDIN_ADS and got rid of a few places where we also mentioned test which we should avoid
- I broke apart some of the bigger blocks of granting permissions into individual bullet points so that we could annotate it more. The comments can make it hard to tell whats it’s doing
- Removed the under the hood information about calling the stored procedure. I don’t think thats necessary for users to know or understand

Otherwise the rest looks great. Nice work and approved!

**Review by brianjlai — APPROVED** [review]

File: docs/snowflake-native-apps/linkedin-ads.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# LinkedIn Ads Connector
+
+The Airbyte connector for LinkedIn Ads is a Snowflake Native Application that allows you to extract data from your LinkedIn Ads account and load records into a database of your choice within Snowflake.
```
Comment (katmarkham) [inline docs/snowflake-native-apps/linkedin-ads.md:3]:
```suggestion
The LinkedIn Ads Connector by Airbyte is a Snowflake Native Application that allows you to extract data from your LinkedIn Ads account and load records into a Snowflake database of your choice.
```

File: docs/snowflake-native-apps/linkedin-ads.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+## Prerequisites
+A LinkedIn Ads account with permission to access data from accounts you want to sync.
+
+## Installing the App
+
+> **Warning**  
+> Do not refresh the Apps page while the application is being installed. This may cause installation to fail.
+
+1. Log into your Snowflake account.
+2. On the left sidebar, click Marketplace.
```
Comment (katmarkham) [inline docs/snowflake-native-apps/linkedin-ads.md:21]:
```suggestion
2. On the left sidebar, click `Marketplace`.
```

File: docs/snowflake-native-apps/linkedin-ads.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+3. Search for `Linkedin Ads Connector` by Airbyte or navigate to https://app.snowflake.com/marketplace/listing/GZTYZ9BCRSJ/airbyte-linkedin-ads-connector-by-airbyte
+4. Click `Request`. This will send a request that we will manually service as soon as we can.
+5. On the left sidebar, click `Apps`.
+6. Under the `Recently Shared with You` section, you should see the `Linkedin Ads Connector by Airbyte`. Click `Get`.
+7. Expand `Options`.
+    1. You can rename the application or leave the default. This is how you will reference the application from a worksheet
```
Comment (katmarkham) [inline docs/snowflake-native-apps/linkedin-ads.md:27]:
```suggestion
    1. You can rename the application or leave the default. This is how you will reference the application from a worksheet.
```

File: docs/snowflake-native-apps/linkedin-ads.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    2. Specify the warehouse that the application will be installed to.
+8. Click `Get`.
+9. Wait for the application to install. Once complete, the pop-up window should automatically close.
+
+You should now see the application under `Installed Apps`. You may need to refresh the page.
```
Comment (katmarkham) [inline docs/snowflake-native-apps/linkedin-ads.md:32]:
```suggestion
You should now see the LinkedIn Ads Connector by Airbyte application under `Installed Apps`. You may need to refresh the page.
```

File: docs/snowflake-native-apps/linkedin-ads.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+## LinkedIn Ads Account
+In order for the Snowflake Native App to query LinkedIn Ads, you will need an account with the right permissions. Please follow [the LinkedIn Ads authentication guide](https://docs.airbyte.com/integrations/sources/linkedin-ads/#set-up-linkedin-ads-authentication-airbyte-open-source) for further information.
+
+## Snowflake Native App Authorizations
+> **Note**  
+> By default the app will be installed using the name `AIRBYTE_LINKEDIN_ADS`, but if you renamed the app during installation, you will have to use that name as a reference.
+
+1. Create the database where the app will access the authorization. This database can be different from the database where the sync will output the records.
```
Comment (katmarkham) [inline docs/snowflake-native-apps/linkedin-ads.md:41]:
```suggestion
1. Create the database where the app will access the authorization. This database can be different from the database where the sync will output records.
```

File: docs/snowflake-native-apps/linkedin-ads.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+CREATE DATABASE <database>;
+USE <database>;
+```
+
+2. The native app will validate the output database and create it if it does not exist. In order to do that, the app needs access to the database:
+```
+GRANT CREATE DATABASE ON ACCOUNT TO APPLICATION <app_name>;
+```
+
+3. We also need to allow outgoing network traffic based on the domain of the source. In the case of LinkedIn Ads, simply run:
```
Comment (katmarkham) [inline docs/snowflake-native-apps/linkedin-ads.md:52]:
```suggestion
3. You will need to allow outgoing network traffic based on the domain of the source. In the case of LinkedIn Ads, simply run:
```

File: docs/snowflake-native-apps/linkedin-ads.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+CREATE OR REPLACE NETWORK RULE linkedin_apis_network_rule
+  MODE = EGRESS
+  TYPE = HOST_PORT
+  VALUE_LIST = ('api.linkedin.com', 'www.linkedin.com', 'linkedin.com');
+```
+
+> **Note**  
+> As of 2023-09-13, the [Snowflake documentation](https://docs.snowflake.com/en/sql-reference/sql/create-external-access-integration) mentions that direct external access is a preview feature and that it is `available to all accounts on AWS` which might restrict the number of users able to use the connector.
+
+4. Once we have external access configured, we need define our authorization/authentication. Provide the credentials to the app as such:
```
Comment (katmarkham) [inline docs/snowflake-native-apps/linkedin-ads.md:63]:
```suggestion
4. Once you have external access configured, you need define your authorization/authentication. Provide the credentials to the app as such:
```

File: docs/snowflake-native-apps/linkedin-ads.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+  SECRET_STRING = '{
+    "auth_method": "oAuth2.0",
+    "client_id": <client_id>,
+    "client_secret": <client_secret>,
+    "refresh_token": <refresh_token>
+  }';
+```
+... where `client_id`, `client_secret` and `refresh_token` are strings. For more information, see [the LinkedIn Ads authentication guide](https://docs.airbyte.com/integrations/sources/linkedin-ads/#set-up-linkedin-ads-authentication-airbyte-open-source).
+
+5. Once the network rule and the secret is defined in Snowflake, we just need to make them available to the app by using an external access integration.
```
Comment (katmarkham) [inline docs/snowflake-native-apps/linkedin-ads.md:76]:
```suggestion
5. Once the network rule and the secret are defined in Snowflake, you need to make them available to the app by using an external access integration.
```

File: docs/snowflake-native-apps/linkedin-ads.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+
+## Configure a connection
+Once this is all set up, you can now configure a connection. To do so, use the Streamlit app by going in the `Apps` section and selecting `AIRBYTE_LINKEDIN_ADS`. You will have to accept the Anaconda terms in order to use Streamlit.
+Once you have access to the app, select `New Connection` and fill the fields as such:
+* Secret: `<database>.<your_schema>.integration_linkedin_ads_oauth`
+* External Access Integration: `integration_linkedin_ads`
+* start_date: UTC date in the format 2020-09-17. Any data before this date will not be replicated.
+* account_ids: Specify the account IDs separated by a space, to pull the data from. Leave empty, if you want to pull the data from all associated accounts. See the [LinkedIn Ads docs](https://www.linkedin.com/help/linkedin/answer/a424270/find-linkedin-ads-account-details) for more info.
```
Comment (katmarkham) [inline docs/snowflake-native-apps/linkedin-ads.md:103]:
```suggestion
* account_ids: Leave empty, if you want to pull the data from all associated accounts. To specify individual account IDs to pull data from, separate them by a space. See the [LinkedIn Ads docs](https://www.linkedin.com/help/linkedin/answer/a424270/find-linkedin-ads-account-details) for more info.
```

File: docs/snowflake-native-apps/linkedin-ads.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+* Output Database: The database where the records will be saved. Snowflake's database naming restriction applies here.
+* Output Schema: The table where the schema will be saved. Snowflake's table naming restriction applies here. 
+* Connection name: How the connection will be referred in the Streamlit app
+* Replication Frequency: How often do the records will be fetched
```
Comment (katmarkham) [inline docs/snowflake-native-apps/linkedin-ads.md:107]:
```suggestion
* Replication Frequency: How often records are fetched
```

File: docs/snowflake-native-apps/linkedin-ads.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+CREATE OR REPLACE SECRET integration_linkedin_ads_oauth
+  TYPE = GENERIC_STRING
+  SECRET_STRING = '{
+    "auth_method": "oAuth2.0",
+    "client_id": <client_id>,
+    "client_secret": <client_secret>,
+    "refresh_token": <refresh_token>
+  }';
+```
+... where `client_id`, `client_secret` and `refresh_token` are strings. For more information, see [the LinkedIn Ads authentication guide](https://docs.airbyte.com/integrations/sources/linkedin-ads/#set-up-linkedin-ads-authentication-airbyte-open-source).
```
Comment (katmarkham) [inline docs/snowflake-native-apps/linkedin-ads.md:74]:
```suggestion
... where `client_id`, `client_secret` and `refresh_token` are strings. For more information, see the [LinkedIn Ads authentication guide](https://docs.airbyte.com/integrations/sources/linkedin-ads/#set-up-linkedin-ads-authentication-airbyte-open-source).
```

**Review by katmarkham — APPROVED** [review]
This layout is much easier to digest! Thanks @maxi297

**Pull request merged**